library/std/src/macros.rs

```diff
@@ -40,6 +40,8 @@ macro_rules! panic {
 /// Use `print!` only for the primary output of your program. Use
 /// [`eprint!`] instead to print error and progress messages.
 ///
+/// This macro does not support writing to non-blocking stdout.
```

Comment:
Could you move this to the `# Panics` section below, and use the language from `print_to`?

Likewise for the other macros.